### ISSUE TITLE
Laravel 5.1 polymorphic whereHas, only trashed.

### ISSUE THREAD

**Issue by erdemkeren** [laravel/framework#13198]
````
dd($self->bar()->onlyTrashed()->whereHas('qux', function ($query) {
                $query->onlyTrashed()->where('foo', 'bar');
})->toSql());
````

Generates:
````
select * from `bars` where `bars`.`owner_id` = ? and `bars`.`owner_id` is not null and `bars`.`owner_type` = ? and `bars`.`deleted_at` is not null and (select count(*) from `quxes` where `bars`.`qux_id` = `quxes`.`id` and `quxes`.`deleted_at` is not null and `foo` = ? and `quxes`.`deleted_at` is null) >= 1
````

To be clear;

> `quxes`.`deleted_at` is not null and `quxes`.`deleted_at` is null

Looks like a bug to me.

**Comment by themsaid** [comment]
Fixed via the PR.

**Comment by ryun** [comment]
@themsaid This is still not fixed for 5.1
